.travis.yml

```diff
@@ -2,12 +2,17 @@ language: go
 go:
 - "1.11"
 
-before_script:
-- go get golang.org/x/tools/cmd/cover
-- go get github.com/mattn/goveralls
-- npm install --verbose ws minimist
+services:
+- docker
 
 script:
-- go get -v -tags netgo ./cmd/ndt-cloud-client
-- go test -v -coverprofile=ndt.cov -coverpkg=./... -tags netgo ./...
-- $GOPATH/bin/goveralls -coverprofile=ndt.cov -service=travis-ci
+- docker build -f TestDockerfile -t ndt-server-test .
+- docker run -t
+    -e TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST
+    -e TRAVIS_JOB_ID=$TRAVIS_JOB_ID
+    -e TRAVIS=true
+    -e COVERALLS_SERVICE_NAME=travis-ci
```

Comment:
IIRC also the `TRAVIS_BRANCH` variable is useful. If coverage is being computed, also `CODECOV_TOKEN` or `COVERALLS_TOKEN` (IIRC) _should_ probably be forwarded to the container. However, perhaps the `COVERALLS_SERVICE_NAME` serves the same purpose here?

Reply:
I mostly just used an existing example that worked. It looks like coveralls and travis are both happy with the current set of flags?

Reply:
Definitely.